library/dicom_frame.php

```diff
@@ -26,119 +29,471 @@
     $type = '.zip';
 }
 
-$web_path = attr($web_path) . '&retrieve&patient_id=' . attr_url($patid) . '&document_id=' . attr_url($docid) . '&as_file=false&type=' . attr_url($type);
-
+if ($web_path) {
+    $csrf = attr_url(CsrfUtils::collectCsrfToken());
```

Comment:
should just use attr() on this since is should not be url encoded (since used in below html input and not the web_path)

Reply:
This is turned around and past in a fetch post. Originally was destined for a GET query so you're prob right.